algebra-utils/src/msm/variable_base.rs

```diff
@@ -160,22 +168,120 @@ impl VariableBaseMSM {
         }) + lowest
     }
 
+    fn msm_inner_cpu<G>(
+        bases: &[G],
+        scalars: &[<G::ScalarField as PrimeField>::BigInt]
+    ) -> G::Projective
+    where
+        G: AffineCurve,
+        G::Projective: ProjectiveCurve<Affine = G>
+    {
+        #[cfg(not(feature = "msm_affine"))]
```

Comment:
We should always use the msm affine since it's faster, without keeping this feature no ?
The only thing is that for TwistedEdwards curves we didn't implement the add_points() function, but rather than introducing another feature that we (should) never use, we should implement the add_points() function instead (even if it's a trivial implementation (that we can put as default implementation in the trait))

Reply:
Yes, of course, if we will always use msm_affine, it should be left as the only CPU-based implementation. Can do this already, or add_points() for TwistedEdwards should be done first?

Reply:
The add_points() function should be an optimization. If it exists and you are able to do it also for TwistedEdwards curves fine, otherwise, for the moment (since we are not going to work with TE curves), you can put a default implementation in the AffineCurve trait that simply does something like: let mut result = G::zero(); for p in points { result += p }.

Reply:
Actually, the CPU implementation of MSM selects the appropriate algorithm depending on the curve and window size. The default should be the msm_affine.